src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas1/Cas1SpaceSearchTest.kt

```diff
@@ -0,0 +1,106 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.cas1
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.ApType
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.Cas1SpaceSearchParameters
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.Cas1SpaceSearchRequirements
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.Cas1SpaceSearchResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.Cas1SpaceSearchResults
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.InitialiseDatabasePerClassTestBase
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.givens.`Given a User`
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.ApprovedPremisesEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.cas1.Cas1SpaceSearchResultsTransformer
+import java.time.LocalDate
+
+class Cas1SpaceSearchTest : InitialiseDatabasePerClassTestBase() {
+  @Autowired
+  lateinit var transformer: Cas1SpaceSearchResultsTransformer
+
+  @Test
+  fun `Search for Spaces without JWT returns 401`() {
+    webTestClient.post()
+      .uri("/cas1/spaces/search")
+      .exchange()
+      .expectStatus()
+      .isUnauthorized
+  }
+
+  @Test
+  fun `Search for Spaces returns OK with correct body`() {
```

Comment:
i'd love to see more coverage of various criteria filtering here but i also appreciate this is MVP and we want to get something stable and working over the next few weeks. I'm not sure on the testing strategy so far for M&M v2 - is the approach to return later to adding wider coverage once things are stable?

Reply:
The endpoint isn't really 'finished' at the moment and will need revisiting (at the very least to implement space availability, but possibly also adjustments to filtering), so from my perspective I'd rather come back to this when it's more complete.